provider/provider.go

```diff
@@ -63,23 +73,27 @@ func NewPrioritizedProvider(priorityCids KeyChanFunc, otherCids KeyChanFunc) Key
 	return func(ctx context.Context) (<-chan cid.Cid, error) {
 		outCh := make(chan cid.Cid)
 
+		priorityCh, err := priorityCids(ctx)
+		if err != nil {
+			return nil, err
+		}
+		otherCh, err := otherCids(ctx)
+		if err != nil {
+			return nil, err
+		}
+
```

Comment:
Instead of only having two `KeyChanFunc` parameters, what if you took a variadic argument:
```go
func NewPrioritizedProvider(cidsFuncs ...KeyChanFunc) KeyChanFunc {
```

Then you iterate through these calling `handleStream` on each:
```go
last := len(cidsFuncs) - 1
for i, cidsFunc := range cidsFuncs {
	cidsCh, err := cidsFunc(ctx)
	if err != nil {
		return nil, err
	}
	handleStream(cidsCh, i < last)
}
```

Reply:
https://github.com/ipfs/boxo/pull/995/commits/4684512c9f3b4d0ba5ece04988e91503a774bb6e